hr_public_holidays/models/hr_public_holidays.py

```diff
@@ -0,0 +1,175 @@
+# ©  2015 2011,2013 Michael Telahun Makonnen <mmakonnen@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import datetime
+from datetime import date
+
+from odoo import fields, models, api, _
+from odoo.exceptions import Warning as UserError
+
+
+class HrPublicHolidays(models.Model):
+    _name = 'hr.holidays.public'
+    _description = 'Public Holidays'
+    _rec_name = 'year'
+    _order = "year"
+
+    display_name = fields.Char(
+        "Name",
+        compute="_compute_display_name",
+        readonly=True,
+        store=True
+    )
+    year = fields.Integer(
+        "Calendar Year",
+        required=True,
+        default=date.today().year
+    )
+    line_ids = fields.One2many(
+        'hr.holidays.public.line',
+        'year_id',
+        'Holiday Dates'
+    )
+    country_id = fields.Many2one(
+        'res.country',
+        'Country'
+    )
+
+    @api.model
+    def create(self, values):
+        holiday = super(HrPublicHolidays, self).create(values)
+        for line in holiday.line_ids:
+            leave = self.env['resource.calendar.leaves'].create(
+                {'name': line.name,
+                 'date_from': datetime.datetime.combine(
+                     fields.Datetime.from_string(line.date),
+                     datetime.time.min),
+                 'date_to': datetime.datetime.combine(
+                     fields.Datetime.from_string(line.date),
+                     datetime.time.max)})
+            line.calendar_leave_id = leave.id
+        return holiday
+
+    @api.multi
+    def write(self, vals):
+        holiday = super(HrPublicHolidays, self).write(vals)
+        for line in holiday.line_ids:
+            line.calendar_leave_id.unlink()
+            leave = self.env['resource.calendar.leaves'].create(
+                {'name': line.name,
+                 'date_from': datetime.datetime.combine(
+                     fields.Datetime.from_string(line.date),
+                     datetime.time.min),
+                 'date_to': datetime.datetime.combine(
+                     fields.Datetime.from_string(line.date),
+                     datetime.time.max)})
+            line.calendar_leave_id = leave.id
+        return holiday
+
+    @api.onchange('line_ids')
+    def _onchange_line_ids(self):
+        for line in self.line_ids:
+            line.calendar_leave_id.unlink()
+            leave = self.env['resource.calendar.leaves'].create(
```

Comment:
I don't think that unlink() and create() methods called inside an onchange are a good idea. If you adopt the workflow I proposed (see https://github.com/OCA/hr/pull/401/files#r160138849), you will not need this onchange method at all.